include/PROPOSAL/scattering/ScatteringMultiplier.h

```diff
@@ -7,12 +7,12 @@ class ScatteringMultiplier : public Scattering {
     double multiple_scatt = 1;
     std::vector<std::pair<InteractionType, double>> stochastic_deflect;
 
-    std::array<double, 2> _scale_deflect(std::array<double, 2>& angles, InteractionType t) override
+    DirectionChangeAngular _scale_deflect(DirectionChangeAngular& angles, InteractionType t) override
     {
         for (auto m : stochastic_deflect) {
             if (m.first == t) {
-                std::get<0>(angles) *= m.second;
-                assert(std::get<0>(angles) <= PI);
+                angles.zenith *= m.second;
+                assert(angles.zenith <= PI);
```

Comment:
Why is the assert here? these are not evaluated in optimized compiled code.

Reply:
Actually I don't know why here is an assert because I haven't implemented it, but instead we can use: 
`if (angles.zenith >= PI) { angles.zenith = 2*PI - angles.zenith};`

Reply:
zenith is not defined in [0, 2π], only in [0, π] and a zenith larger than π is an error (or if small, due to floating point precision).

Reply:
That's correct and therefore I suggested to clip the angle. To be completely correct here we should have to check if zenith is higher than PI and smaller than 2PI. 
It is not obvious what to do with scaled angles which are higher than PI because in the most cases the deflection angles are very small but there are a few deflections which can reach quite high angles. Maybe we can set scaled angles to PI if they are higher.

Reply:
> Why is the assert here? these are not evaluated in optimized compiled code.

In general, a multiplier that produces angles greater than pi is too large. For studies, only multipliers that will not go outside the definition range should be used and this should be checked by the user before. The assert exists only to simplify the debugging. 

I think there is no possibility to trade deflection angles larger than 180 degrees. The consistent possibility is to throw an assert that the angle distribution is broken.